File: integration-tests/mongodb-panache/src/main/java/io/quarkus/it/mongodb/panache/person/PersonEntity.java
package io.quarkus.it.mongodb.panache.person;

import java.util.List;

import org.bson.codecs.pojo.annotations.BsonId;

import io.quarkus.mongodb.panache.PanacheMongoEntityBase;
import io.quarkus.panache.common.Sort;

public class PersonEntity extends PanacheMongoEntityBase {
    @BsonId
    public Long id;
    public String firstname;
    public String lastname;

    public static List<PersonEntity> findOrdered() {
        return findAll(Sort.by("lastname", "firstname")).list();
    }
}

File: integration-tests/mongodb-panache-kotlin/src/main/kotlin/io/quarkus/it/mongodb/panache/person/PersonEntity.kt
package io.quarkus.it.mongodb.panache.person

import io.quarkus.mongodb.panache.kotlin.PanacheMongoCompanionBase
import io.quarkus.mongodb.panache.kotlin.PanacheMongoEntityBase
import io.quarkus.panache.common.Sort
import org.bson.codecs.pojo.annotations.BsonId

class PersonEntity : PanacheMongoEntityBase() {
    @BsonId
    var id: Long? = null
    var firstname: String? = null
    var lastname: String? = null
    var status: Status = Status.ALIVE

    companion object : PanacheMongoCompanionBase<PersonEntity, Long> {
        fun findOrdered(): List<PersonEntity> {
            return findAll(Sort.by("lastname", "firstname")).list()
        }

    }
}